bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/handler/LGWebOSTVSocket.java

```diff
@@ -332,6 +333,10 @@ private void sendMessage(JsonObject json) {
     @OnWebSocketMessage
     public void onMessage(String message) {
         logger.trace("Message [in]: {}", message);
+        if (this.listener == null) {
+            logger.debug("Message is ignored because the thing handler is already disposed.");
+            return;
+        }
```

Comment:
Is there a more direct way of determining if the handler is disposed? Or more importantly, shouldn't this class be disposed along with the handler in the first place?

Reply:
In fact, this is the more direct way I found.
Here is the code run when the thing handler is disposed:
```
        LGWebOSTVSocket s = socket;
        if (s != null) {
            s.setListener(null);
            scheduler.execute(() -> s.disconnect()); // dispose should be none-blocking
        }
```
As you can see, this is done asynchronously. It can happen that messages are received and handled bedore `disconnect()` is run.

setListener is called with the thing handler in thing handler initialize().
setListener is called with null in thing handler dispose().

Reply:
The variable is named `listener` but in fact listener = thingHandler

Reply:
I'm not sure that `s.disconnect()` should be called asynchronously as part of disposing. Any essential cleanup should be done immediately because `initialize()` might be called immediately afterward and any socket resources getting initialized might collide with the socket resources that are getting disposed.

But I'm not sure if such a change is within the scope of this PR...

Reply:
@sprehn : can you comment about the asynchronous disconnect please ? The comment looks clear, is s.disconnect() really blocking ? Until what ?

@cpmeister : yes, the scope of my PR was not to change that but rather avoid having the reconnect job continuing its scenario while the thing handler is already disposed.

Reply:
yes we indeed can make this call synchronous 
s.disconnect enqueues a websocket close, which is asynchronous 
it synchronously sets the state to disconnected

Reply:
the lgweostvsocket instance gets disposed 
a new instance is created during initialize

can you remind me what the issues is that needs to be solved?

Reply:
You can read again last @cpmeister message, he explained the problem. It occurs when the thing handler is quickly disposed and initialized again while the first registering process is still running.

Reply:
but we create a new socket each time we initialize. so states should be clean. 

what is the issue for end users?